From 8fc10c02f53754dfd83cf50327e801cdec4d2895 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabiano=20Fid=C3=AAncio?= <fidencio@redhat.com>
Date: Thu, 9 Oct 2014 17:05:05 +0200
Subject: [PATCH] Adapt auth_cancelled to the downstream code

While upstream we have virt_viewer_auth_collect_credentials() returning
a boolean (FALSE, 0, indicating that the user cancelled the auth dialog
and TRUE, !0, otherwise), downstream the function is still returning an
integer (-1 indicating that the user cancelled the auth dialog and 0
otherwise).

Related: rhbz#1142742
---
 src/virt-viewer.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/virt-viewer.c b/src/virt-viewer.c
index 604dea6..4159203 100644
--- a/src/virt-viewer.c
+++ b/src/virt-viewer.c
@@ -693,13 +693,15 @@ virt_viewer_auth_libvirt_credentials(virConnectCredentialPtr cred,
     if (username || password) {
         VirtViewerWindow *vwin = virt_viewer_app_get_main_window(VIRT_VIEWER_APP(app));
         GtkWindow *win = virt_viewer_window_get_window(vwin);
-        priv->auth_cancelled = !virt_viewer_auth_collect_credentials(win,
-                                                                     "libvirt",
-                                                                     app->priv->uri,
-                                                                     username, password);
-        if (priv->auth_cancelled) {
-            ret = -1;
+        ret = virt_viewer_auth_collect_credentials(win,
+                                                   "libvirt",
+                                                   app->priv->uri,
+                                                   username, password);
+        if (ret < 0) {
+            priv->auth_cancelled = TRUE;
             goto cleanup;
+        } else {
+            priv->auth_cancelled = FALSE;
         }
     }
 
